### PR TITLE
[milestone][m]: added option for bot issue to assign into milestone -…

### PR DESCRIPTION
**Pull request by Mikanebu**
- refactored current bot issue functions
- created new function `listMilestones` since we can use only `milestoneNumber` I had to make a API call by using `milestoneTitle`
- added unittest

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,7 +31,7 @@
     "hubot-shipit": "^0.2.0",
     "hubot-timezone": "^1.0.1",
     "moment": "^2.18.1",
-    "node-gitter": "^2.0.5",
```
Comment (zelima) [inline package.json:34]: Why are we fixing this version?

**Review by zelima — APPROVED** [review]
Seems fine, just a small comment

File: package.json (hunk 1 above)
Comment (Mikanebu) [reply package.json:34]: @zelima yesterday, they did some commit and released new version `2.1.0` but it errored for me, since it will look for `token`.

Comment (zelima) [reply package.json:34]: @Mikanebu for what token? Maybe it's worth updating our codebase if it's something simple. New version means, some bugs are fixed or some new feature is released, by fixing it we will not have access to new features of it.

Comment (Mikanebu) [reply package.json:34]: @zelima There is no issue fixed or updates on readme.md. This is a piece of code added.
```
  token = token.trim();
 +  
 +  if (token.length !== 40) {
 +    console.warn('Token may not be valid, ' + token + '. Tokens are usually 40 character hex strings.');
 +  }
```

Comment (zelima) [reply package.json:34]: That's interesting why did they add this snippet. But ok let's merge for now and investigate later

Comment (Mikanebu) [reply package.json:34]: @zelima ok

**Pull request merged**